Content.Shared/Clothing/Components/Clown/Waddle.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System.Numerics;
+
+namespace Content.Shared.Clothing.Components.Clown;
+
+/// <summary>
+/// Defines something as causing waddling when worn.
+/// </summary>
+[RegisterComponent]
+public sealed partial class WaddleComponent : Component
+{
+    ///<summary>
+    /// How high should they hop during the waddle? Higher hop = more energy.
+    /// </summary>
+    [DataField]
+    public Vector2 HopIntensity = new(0, 0.25f);
+
+    /// <summary>
+    /// How far should they rock backward and forward during the waddle?
+    /// Each step will alternate between this being a positive and negative rotation. More rock = more scary.
+    /// </summary>
+    [DataField]
+    public float TumbleIntensity = 20.0f;
+
+    /// <summary>
+    /// How long should a complete step take? Less time = more chaos.
+    /// </summary>
+    [DataField]
+    public float AnimationLength = 0.75f;
```

Comment:
Unless I'm missing something, these values are never actually copied over to the animation component, so it will always use the default values.

I might be getting carried away a bit here, but instead of duplicating three fields manually (and making sure to update the copying logic if more fields are added in the future) it might be cleaner to define a WaddlePrototype that has all the needed fields and just copy the ProtoId over to the animation component.

Reply:
This would need me walked through it by someone, potentially one of the contrib voice chatters. Does it block an approval itself or is there something simpler I could do to get this over the line?

Reply:
I'd be fine with just copying each field over instead of doing the whole prototype thing.